src/web/components/SlotBodyEnd/BrazeEpic.tsx

```diff
@@ -0,0 +1,169 @@
+	// logButtonClickWithBraze: (id: number) => void; // TODO: Handle button click tracking
+};
+
+type EpicConfig = {
+	meta: Meta;
+	contributionsServiceUrl: string;
+	countryCode: string;
+};
+
+type Tracking = {
+	ophanPageId: string;
+	platformId: string;
+	clientName: string;
+	referrerUrl: string;
+};
+
+type EpicProps = {
+	variant: Variant;
+	tracking: Tracking;
+	numArticles: number;
+	countryCode: string;
+};
+
+export const canShow = async (
+	brazeMessagesPromise: Promise<BrazeMessagesInterface>,
+): Promise<CanShowResult> => {
+	try {
+		const brazeMessages = await brazeMessagesPromise;
+		const message = await brazeMessages.getMessageForEndOfArticle();
+
+		return {
+			result: true,
+			meta: {
+				dataFromBraze: message.extras,
+				logImpressionWithBraze: () => {
+					message.logImpression();
+				},
+			},
+		};
+	} catch (e) {
+		return { result: false };
+	}
+};
+
+const buildEpicProps = (
+	dataFromBraze: EpicDataFromBraze,
+	countryCode: string,
+): EpicProps | null => {
+	const variant = parseBrazeEpicParams(dataFromBraze);
+	if (variant === null) return null;
+
+	const pageTracking = {
+		ophanPageId: window.guardian.config.ophan.pageViewId,
+		platformId: 'GUARDIAN_WEB',
+		clientName: 'dcr',
+		referrerUrl: window.location.origin + window.location.pathname,
+	};
+
+	const tracking = {
+		...pageTracking,
+	};
+
+	return { variant, tracking, numArticles: 0, countryCode };
+};
+
+const BrazeEpic = ({
+	contributionsServiceUrl,
+	meta,
+	countryCode,
+}: EpicConfig) => {
+	const [Epic, setEpic] = useState<React.FC<EpicProps>>();
+
+	useEffect(() => {
+		window.guardian.automat = {
+			react: React,
+			preact: React,
+			emotionCore,
+			emotionTheming,
+			emotion,
+		};
+		const componentUrl = `${contributionsServiceUrl}${EPIC_COMPONENT_PATH}`;
+		window
+			.guardianPolyfilledImport(componentUrl)
+			.then((epicModule: { ContributionsEpic: React.FC<EpicProps> }) => {
+				setEpic(() => epicModule.ContributionsEpic); // useState requires functions to be wrapped
+				// TODO: log the impression when the epic is in view (using hasBeenSeen probably)
+				meta.logImpressionWithBraze();
+			})
+			// eslint-disable-next-line no-console
+			.catch((error) => console.log(`epic - error is: ${error}`));
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	if (Epic && meta.dataFromBraze) {
+		// This will come from Braze via the meta from canShow
+		const props = buildEpicProps(
+			meta.dataFromBraze as EpicDataFromBraze,
+			countryCode,
+		);
+
+		if (props) {
+			return (
+				<div className={wrapperMargins}>
+					{/* eslint-disable-next-line react/jsx-props-no-spreading */}
+					<Epic {...props} />
+				</div>
+			);
+		}
+		console.log('End of article props were not defined.');
+	}
+
+	return null;
+};
+
+const NoEpic = () => null;
+
+export const MaybeBrazeEpic = ({
+	contributionsServiceUrl,
+	meta,
+	countryCode,
+}: EpicConfig) => {
+	const componentName = meta.dataFromBraze?.componentName;
```

Comment:
I suspect this code might be from another PR so this is more a nitpick but I find the name `componentName` confusing because it's not actually the name of a component 🤷

Reply:
Yeah you're right, this is a name we already use for the banner where it actually _is_ a component name (from `@guardian/braze-components`. That mapping isn't quite the same here because we're piggybacking on the reader revenue epic component from support-dotcom-components.

Reply:
Not a huge issue and not worth refactoring for unless it's easy